clang/lib/Driver/Driver.cpp

```diff
@@ -3322,6 +3438,16 @@ class OffloadingActionBuilder final {
               C.getInputArgs().getLastArg(options::OPT_fsycl_targets_EQ);
       bool HasValidSYCLRuntime = C.getInputArgs().hasFlag(options::OPT_fsycl,
                                               options::OPT_fno_sycl, false);
+
+      // Add additional check for -fintelfpga, which if set will add the fpga
+      // specific target triple.  There is a check done earlier for the setting
+      // of both options, so we can freely just check argument contents.
+      if (!SYCLTargets && C.getInputArgs().hasArg(options::OPT_fintelfpga)) {
+        const char * AStr =
+                Args.MakeArgString("spir64_fpga-unknown-linux-sycldevice");
+        SYCLTargets = Args.MakeJoinedArg(nullptr,
+        C.getDriver().getOpts().getOption(options::OPT_fsycl_targets_EQ), AStr);
```

Comment:
Indentation seems a little off here

Reply:
I tried to keep formatting within 80 columns which is why it is at the same level.  I'll do some slight refactoring.